### PR TITLE
Add default jobname and raise more descriptive config validation exception

### LINKED ISSUE #4174: Add separate exception handling of config checking
Add better handling of config validation

### PR DESCRIPTION
**Pull request by andreas-el**
Resolves #4174 and adds the potentially breaking change of default jobnames.

There will be squashing later on.
- Is the placement of the exception acceptable? I had some issues with circular dependencies on the original placement.
- Verify label

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/res_config.py [hunk 1]
```diff
@@ -182,17 +182,21 @@ def _alloc_from_content(self, user_config_file=None, config=None):
...
+                        "<CWD>": self.config_path,
+                        "<CONFIG_PATH>": self.config_path,
+                        "<CONFIG_FILE>": os.path.basename(user_config_file),
+                        "<CONFIG_FILE_BASE>": os.path.basename(user_config_file).split(
+                            "."
+                        )[0],
+                    },
+                )
+            except ValueError as err:
+                raise ConfigValidationError(err)
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/res_config.py:199]: This is not really what i wanted to do, rather i'd like to raise `ConfigValidationError` directly from config_parser.py:
```python
        if validate and not config_content.isValid():
            sys.stderr.write(f"Errors parsing:{config_file} \n")
            for count, error in enumerate(config_content.getErrors()):
                sys.stderr.write(f"  {count:02d}:{error}\n")
            raise ValueError(f"Parsing:{config_file} failed")

        return config_content
```
My problem is that doing so causes a circular dependency somehow. I will look further into this.

**Comment by andreas-el** [comment]
Jenkins retest this please!

File: src/ert/_c_wrappers/enkf/res_config.py (hunk 1 above)
Comment (andreas-el) [reply src/ert/_c_wrappers/enkf/res_config.py:196]: Aaand it's done! Doomguy says so. :suspect:

Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/res_config.py]: I am guessing you moved `ConfigValidationError` from `res_config.py` to `config_parser.py` which is reasonable :) . I guess alternatively we might have wanted to to have a separate file for `ConfigValidationError`, but I think this is also good.

**Review by eivindjahren — APPROVED** [review]
If you don't disagree, I added the breaking change as the first item in the PR title and added the breaking change label. Probably, very few users (if any) will be notice that the default has been added, but I think better safe than sorry in this case. I believe the location of the error is good.

File: src/ert/_c_wrappers/enkf/res_config.py (hunk 1 above)
Comment (andreas-el) [reply src/ert/_c_wrappers/enkf/res_config.py]: Should we separate the exception into a file? That would be super easy.
Also, i'll squash everything to one commit.

**Comment by valentin-krasontovitsch** [comment]
ok great eivind was faster, should have seen it coming XD sorry about my slowness!

**Pull request merged**